Sources/CoreMetrics/Metrics.swift

```diff
@@ -384,9 +386,16 @@ public enum MetricsSystem {
             self._factory = factory
         }
     }
+    
+    internal static var systemMetrics: SystemMetricsFactory? {
+        return self.lock.withReaderLock { _factory as? SystemMetricsFactory }
+    }
 
     /// Returns a reference to the configured factory.
     public static var factory: MetricsFactory {
+        if let f = self.systemMetrics {
+            return f.underlying
+        }
```

Comment:
is this in place to save a hop?

Reply:
The reason I did this is so that for the outside world, `MetricsSystem.factory` will still give back the factory they used.

Not doing this would break behaviour like this (used in `SwiftPrometheus`):
```swift
    static func prometheus() throws -> PrometheusClient {
        guard let prom = self.factory as? PrometheusClient else {
            throw PrometheusError.prometheusFactoryNotBootstrapped(bootstrappedWith: "\(self.factory)")
        }
        return prom
    }
```

This also allows us to keep `SystemMetrics` 100% internal other than passing in the `Configuration` at bootstrap time.